src/engine/plugins/table.cpp

```diff
@@ -127,6 +128,12 @@ Status TablePlugin::HandleRequest(const RoutingAlgorithmsInterface &algorithms,
                         result_tables_pair.second[table_index] = distance_estimate;
                     }
                 }
+                if (params.scale_factor > 1.0 &&
+                    result_tables_pair.first[table_index] != MAXIMAL_EDGE_DURATION)
+                {
+                    result_tables_pair.first[table_index] =
+                        result_tables_pair.first[table_index] * (double)params.scale_factor;
```

Comment:
Should probably check for overflow here, and clamp the result to some very large number. - if value * scale_factor ends up > MAXIMAL_EDGE_DURATION (i.e. overflows), we should set `value` to `MAXIMAL_EDGE_DURATION-1`.

The value of `MAXIMAL_EDGE_DURATION` is reserved to indicate "noroute", so we have to clamp slightly lower than that.

Something like:

```
double result = result_tables_pair.first[table_index] * params.scale_factor;
if (double > MAXIMAL_EDGE_DURATION) {
  result_tables_pair.first[table_index] = MAXIMAL_EDGE_DURATION - 1;
} else {
  result_tables_pair.first[table_index] = result;
}
```

Reply:
Also note - no need to cast `params.scale_factor` to `double` - it's already one of those.  Also, it's preferred in C++ to use `static_cast<double>(thingy)` instead of C-style `(double)thing` casts - the C++ style has better defined behaviour, which makes it easier to debug if something goes wrong.

Reply:
👍

Reply:
@danpat after applying a clamp, should we return an error? Or return the raw really large value? For instance in this test case below, the really large unexplained number gets piped through. 

Also, we need a clamp for a negative multiplier too, yes?

```
 ✔ Given the profile "testbot"
  ✔ And the partition extra arguments "--small-component-size 1 --max-cell-sizes 2,4,8,16"
  ✔ Given the query options
      | scale_factor | 2147483647 |
  ✔ Given the node map
      """
      a b
      """
  ✔ And the ways
      | nodes |
      | ab    |
  ✖ When I request a travel time matrix I should get
      |   | a  | b  |
      | a | 0  | 20 |
      | b | 20 | 0  |

Failures:

1) Scenario: Testbot - Travel time matrix of minimal network with scale factor - features/testbot/duration_matrix.feature:611
   Step: When I request a travel time matrix I should get - features/testbot/duration_matrix.feature:624
   Step Definition: features/step_definitions/distance_matrix.js:10
   Message:
     Tables were not identical:
     |   |     a  |     b  |
     | a |     0  | (-) 20 |
     | a |     0  | (+) 214748364.6 |
     | b | (-) 20 |     0  |
     | b | (+) 214748364.6 |     0  |
```

Reply:
Hm I think we should clamp down the `scale_factor` itself. Try to clamp the result down after an overflow has occured is too late, and to check every element before doing the multiplication to prevent an overflow will be expensive. 

This is the kind of check that would happen before every multiplication:
```
EdgeDuration diff = MAXIMAL_EDGE_DURATION / std::lround(result_tables_pair.first[table_index]);
       if (params.scale_factor > diff) {
            // overflow will happen for std::lround(result_tables_pair.first[table_index]) * params.scale_factor so error
       }
result_tables_pair.first[table_index] =  std::lround(result_tables_pair.first[table_index] * params.scale_factor);
```

Reply:
I've gone with `return the raw really large value`.